pkg/cmd/cli/cmd/set/triggers.go

```diff
@@ -656,8 +656,13 @@ func (t *TriggerDefinition) Apply(obj runtime.Object) error {
 			allNames.Insert(container.Name)
 		}
 		for _, trigger := range t.ImageChange {
+			// If we have only one container, assume the triggers should be set for that
+			// container and do not require --containers to be specified.
+			if len(allNames) == 1 {
+				trigger.Names = []string{allNames.List()[0]}
```

Comment:
If `--containers` is specified and identifies a container that does not exist, will this silently override the user's specification?